File: javastory.budgetsh.stage1/src/javastory/budgetsh/stage1/step3/ui/menu/MainMenu.java
package javastory.budgetsh.stage1.step3.ui.menu;

import java.util.Scanner;

import javastory.budgetsh.stage1.util.Narrator;
import javastory.budgetsh.stage1.util.TalkingAt;

public class MainMenu {
	//
	private AccountMenu accountMenu;
	private CashBookMenu cashbookMenu;
	
	private Scanner scanner;
	private Narrator narrator;
	
	public MainMenu() {
		//
		accountMenu = new AccountMenu();
		cashbookMenu = new CashBookMenu();
		
		scanner = new Scanner(System.in);
		narrator = new Narrator(this, TalkingAt.Left);
	}
	
	public void show() {
		//
		int inputNumber =0;
		
		while(true) {
			//
			displayMenu();
			inputNumber = selectNumber();
			
			switch(inputNumber) {
			//
			case 1:
				cashbookMenu.show();
				break;
			case 2:
				accountMenu.show();
				break;
			case 0:
				exit();
				break;
				
			default:
				narrator.sayln("Choose again!");
			}
		}
	}
	
	public void displayMenu() {
		//
		narrator.sayln("------------------------------");
		narrator.sayln(" Main Menu");
		narrator.sayln("------------------------------");
		narrator.sayln(" 1. CashBook menu");
		narrator.sayln(" 2. Account Menu");
		narrator.sayln("------------------------------");
		narrator.sayln(" 0. exit");
	}
	
	public int selectNumber() {
		//
		int inputNumber = 0;
		inputNumber = scanner.nextInt();
		
		if (inputNumber >= 0 && inputNumber <= 2) {
			scanner.nextLine();
			return inputNumber;
		} else {
			narrator.sayln("It's InvalidNumber -->" + inputNumber);
			return -1;
		}
		
	}
	
	public void exit() {
		//
		narrator.sayln("Exit...");
		System.exit(0);
	}
}

File: javastory.budgetsh.stage5/src/ui/menu/MainMenu.js
var co = require('co');
var prompt = require('co-prompt');
var commander = require('commander');
const CashbookMenu = require('./CashBookMenu.js');
const AccountMenu = require('./AccountMenu.js');

class MainMenu {
	constructor() {
		
	}
	
	show() {
		this.selectNumber();
	}

	displayMenu() {
		console.log("------------------------------");
		console.log(" Main Menu");
		console.log("------------------------------");
		console.log(" 1. CashBook menu");
		console.log(" 2. Account Menu");
		console.log("------------------------------");
		console.log(" 0. exit");
	}

	selectNumber() {
		var selectedNumber = '';
		this.displayMenu();
		co(function*() {
			var selectedNumber = yield prompt('>  ');
			var numberLine = Number(selectedNumber);
			if (numberLine >= 0 && numberLine <= 2) {
				sendToMenu(numberLine);
			}else{
				this.selectNumber();
			}
		});

	}

	exit() {
		console.log('Exit..');
		process.exit();
	}
}
var mainMenu = new MainMenu();

function sendToMenu(selectNumber) {
	switch (selectNumber) {
	case 0:
		mainMenu.exit();
		break;

	case 1:
		var cashbookMenu = new CashbookMenu();
		cashbookMenu.show(mainMenu);
		break;

	case 2:
		var accountMenu = new AccountMenu();
		accountMenu.show(mainMenu);
		break;
	}
}



mainMenu.show();
module.exports = MainMenu;